excel_import_export/wizard/import_xlsx_wizard.py

```diff
@@ -0,0 +1,157 @@
+                            "This import action is not usable "
+                            "in this document context"
+                        )
+                    )
+                    break
+        if not valid:
+            raise ValidationError("\n".join(messages))
+        return res
+
+    @api.model
+    def default_get(self, fields):
+        res_model = self._context.get("active_model", False)
+        res_id = self._context.get("active_id", False)
+        template_domain = self._context.get("template_domain", [])
+        templates = self.env["xlsx.template"].search(template_domain)
+        if not templates:
+            raise ValidationError(_("No template found"))
+        defaults = super().default_get(fields)
+        for template in templates:
+            if not template.datas:
+                act = self.env.ref("excel_import_export.action_xlsx_template")
+                raise RedirectWarning(
+                    _(
+                        'File "%(fname)s" not found in template, %(name)s.',
+                        fname=template.fname,
+                        name=template.name,
+                    ),
+                    act.id,
+                    _("Set Templates"),
+                )
+        defaults["template_id"] = len(templates) == 1 and template.id or False
+        defaults["res_id"] = res_id
+        defaults["res_model"] = res_model
+        return defaults
+
+    def get_import_sample(self):
+        self.ensure_one()
+        return {
+            "name": _("Import Excel"),
+            "type": "ir.actions.act_window",
+            "res_model": "import.xlsx.wizard",
+            "view_mode": "form",
+            "res_id": self.id,
+            "views": [(False, "form")],
+            "target": "new",
+            "context": self._context.copy(),
+        }
+
+    def action_import(self):
+        self.ensure_one()
+        Import = self.env["xlsx.import"]
+        res_ids = []
+        if self.import_file:
+            record = Import.import_xlsx(
+                self.import_file, self.template_id, self.res_model, self.res_id
+            )
+            res_ids = [record.id]
+        elif self.attachment_ids:
+            for attach in self.attachment_ids:
+                record = Import.import_xlsx(attach.datas, self.template_id)
+                res_ids.append(record.id)
+        else:
+            raise ValidationError(_("Please select Excel file to import"))
+        # If redirect_action is specified, do redirection
+        if self.template_id.redirect_action:
+            vals = self.template_id.redirect_action.read()[0]
+            vals["domain"] = [("id", "in", res_ids)]
+            return vals
+        self.write({"state": "get"})
+        return {
+            "type": "ir.actions.act_window",
+            "res_model": self._name,
+            "view_mode": "form",
+            "res_id": self.id,
+            "views": [(False, "form")],
+            "target": "new",
+        }
```

Comment:
The action is missing a name (title), therefore after continuing by executing `action_import()`. The window title is no longer the export's name but `Odoo`.